### DISCUSSION TITLE
Trying to enforce a chair conformation during 3D geometry generation

### DISCUSSION THREAD

**Discussion by metma99** [rdkit/rdkit#7709] [Q&A] [answered]
Hello:

I am trying to reproduce the example shown here:
https://sourceforge.net/p/rdkit/mailman/rdkit-discuss/thread/5fea4cff.3b8d.16c5f6a4643.Coremail.masgils%40163.com/#msg36733323

However, in my case the 6 membered ring is not enforced as chair.

Here is the example:
`
from IPython.display import display, HTML
from rdkit  import  Chem
from rdkit.Chem  import  AllChem
from rdkit.Chem.Draw  import  IPythonConsole

dimethyl_cyclohexyl_core = Chem.MolFromMolBlock('''\
cyclohexyl
     RDKit          3D

  8  8  0  0  0  0  0  0  0  0999 V2000
    1.2630    0.7280   -0.2259 C   0  0  2  0  0  0  0  0  0  0  0  0
    2.4150    1.3926    0.2940 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.0010    1.4580    0.2251 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.2620    0.7290   -0.2259 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.2630   -0.7280    0.2261 C   0  0  1  0  0  0  0  0  0  0  0  0
   -2.4150   -1.3926   -0.2939 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.0010   -1.4580   -0.2249 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.2620   -0.7290    0.2261 C   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  1
  1  3  1  0
  1  8  1  0
  3  4  1  0
  4  5  1  0
  5  6  1  6
  5  7  1  0
  7  8  1  0
M  END
''')

cyclohexyl_core = Chem.MolFromMolBlock('''\
cyclohexyl
     RDKit          3D

  6  6  0  0  0  0  0  0  0  0999 V2000
    1.2630    0.7280   -0.2259 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.0010    1.4580    0.2251 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.2620    0.7290   -0.2259 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.2630   -0.7280    0.2261 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.0010   -1.4580   -0.2249 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.2620   -0.7290    0.2261 C   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  1
  1  6  1  0
  2  3  1  0
  4  3  1  6
  4  5  1  0
  5  6  1  0
M  END
''')

params  =  AllChem.AdjustQueryParameters()
params.makeAtomsGeneric  =  True
dimethylchair_core_generic  =  AllChem.AdjustQueryProperties(dimethyl_cyclohexyl_core,  params)
chair_core_generic  =  AllChem.AdjustQueryProperties(cyclohexyl_core,  params)

bilastine_smi  =  "CCOCCn1c(C2CC[NH+](CCc3ccc(C(C)(C)C(=O)[O-])cc3)CC2)nc2ccccc21"
bilastine = Chem.MolFromSmiles(bilastine_smi)
bilastine = Chem.AddHs(bilastine)

bilastine_cyh  =  AllChem.ConstrainedEmbed(bilastine,  chair_core_generic,
                                         useTethers=True)

IPythonConsole.drawMol3D(bilastine_cyh)

`

How can I enforce the chair conformation?
Any suggestions her highly appreciated,
Markus

**Comment by greglandrum** [comment, marked as answer]
Hi @metma99 
There's nothing built in that will allow you to constrain the geometry to be a boat, but if you tell the conformer generator to use the small ring torsions, you can at least more or less guarantee that you will get either a chair or a boat. You should be able to do this by passing the keyword argument `useSmallRingTorsions=True` to `ConstrainedEmbed()`

**Reply by phonglam3103** [reply]
Hi @greglandrum, just want to ask a follow-up question on this. Is there any way to add multiple templates into ConstrainedEmbed? For example: I want to construct a library of small rings/fragments like cyclohexane, pyran,... which I would like rdkit to build my new molecules based on these favorable building block conformations. Is it possible yet?
Thank you,
/P.

**Reply by greglandrum** [reply]
@phonglam3103 you can only provide one template per molecule.
You can do what you want by figuring out which template matches your molecule (with a substructure search) and then calling `ConstrainedEmbed` with that template.
